### ISSUE TITLE
Forms Demo V2

### ISSUE THREAD

**Issue by rachael-phillips** [patternfly/patternfly-design#858]
Continued revision on Forms Demo based on requirements from patternfly/patternfly#389 

This demo will be presented on the website.

As part of this, we should demo what a full page form layout looks like when we put it into the context of a page layout.  How do we determine the width of a form? Are there different variants?

**Comment by mmenestr** [comment]
Looking at the above designs/conversation, there have been a few updates to the Form design guidelines that would affect these future demos. Here are the main takeaways and screenshots to illustrate what the demos should look like! Let me know if more are needed.

1. The form itself should be laid out the same, but the Cancel button should be a secondary button. Ultimately, the basic form demo should look something like this: 
![image](https://user-images.githubusercontent.com/51165119/86956892-d9316600-c127-11ea-9c90-21be52d4a5cb.png)

2. Popover clarification: 
- It should be the unfilled question circle icon
- **Placement of the popover icon**: it should be placed to the right of the **text label**
- **Color of the popover icon**: grey by default, and black on hover and click
![image](https://user-images.githubusercontent.com/51165119/86957038-1d246b00-c128-11ea-88ed-6d1d0abee312.png)

3. Form in a page:
Per a discussion with @mceledonia while I was updating the Form design guidelines, we concluded that when a page only consists of a form, the form should live on the white page background itself and not be placed on a card. In the design guidelines we also suggest that a good max width for the form inputs would be 880px. So, it should look like this:
![image](https://user-images.githubusercontent.com/51165119/86957269-78565d80-c128-11ea-99d1-11b9e12c6421.png)

@mceledonia @mcarrano can you confirm you agree with all of that^?

**Comment by mcarrano** [comment]
This looks good to me. In addition to the above examples, do you think it's useful to show an example of a form in a modal, @mmenestr ? Maybe we can put all of the demo examples in a Marvel deck so it's easier to follow for implementation? There is so much history here, it becomes hard to follow. Once we have that I will open a clean issue for core implementation of these as demos.

**Comment by mcarrano** [comment]
@mmenestr can you take a look at this issue: https://github.com/patternfly/patternfly/issues/2489 We should make sure that we cover this type of a layout as part of the demo. It's just basically asking for a group of stacked radio buttons, I think. Thanks!

**Comment by mmenestr** [comment]
Here are the demo mocks:
- [Vertical form demo](https://marvelapp.com/prototype/eh3c7d4/screen/71128973)
-[Horizontal form demo](https://marvelapp.com/prototype/eh3c7d4/screen/71128976)
- [Stacked radio buttons demo ](https://marvelapp.com/prototype/eh3c7d4/screen/71129134)(with 8px spacing between buttons)
- [Full page vertical form demo](https://marvelapp.com/prototype/eh3c7d4/screen/71128974)

The main difference between these and what's currently on the site, is the addition of the popovers, the stacked radio buttons, and the new full page demo with no card behind it!

**Comment by mcarrano** [comment]
The mocks look good @mmenestr . I only had one comment. @mceledonia can you also take a look at these?

**Comment by mcarrano** [comment]
@mceledonia @mcoker Please take a look at the mockups attached above. If you are good with these, I will open a new core issues to get this into the queue for implementation. Thanks.

**Comment by mcoker** [comment]
@mcarrano @mmenestr a few questions:

Vertical form - the form field error/help text does not push down the fields below it. However the form component currently does push down content to make room for the error text. Just want to confirm whether or not we need to update anything regarding that?

Stacked radios - would that same 8px gap apply to other things that you should be able to stack in a form - such as checkboxes, text inputs, other form control elements (select, textarea), etc?

Here is what it looks like currently if you stack text inputs associated with a single form label.

<img width="705" alt="Screen Shot 2020-08-04 at 5 34 25 PM" src="https://user-images.githubusercontent.com/35148959/89352045-dff4bf80-d678-11ea-9b84-28191e3f5931.png">

And here is an 8px gap

<img width="710" alt="Screen Shot 2020-08-04 at 5 34 44 PM" src="https://user-images.githubusercontent.com/35148959/89352058-e5eaa080-d678-11ea-9231-19425cb786ff.png">

Full page - the max-width specified in the design is 880px. We just implemented a feature that allows a user to specify a variant of page section that applies a max-width to the content inside (https://github.com/patternfly/patternfly/issues/2910) and that width is 1710px, which comes from `2000px - [the vertical nav/sidebar width, which is 290px]`, which makes the width from the left edge of the browser to the right edge of the content 2000px. Should the form be the same? Or should the form width be different than the data-list, table, text, navigation, breadcrumbs, etc when that max-width variation is applied to a page section?

**Comment by mcarrano** [comment]
@mcoker Just a couple of comments...

* Yes, I think we are wanting to introduce the 8px spacing between any stacked form elements.  I know this came up recently in a separate issue where someone wanted to stack a group of radio buttons.

* Regarding the max-width, yes, we intend the form to have a max-width that is smaller than the page. This will prevent form fields that take the full width of the container from being too stretched out.

**Comment by mmenestr** [comment]
@mcoker +1 on everything Matt said and then regarding the form being pushed down when there's an error - it should stay how it works now - pushing the content below it to keep the spacing consistent with or without extra text. That's just an error in my mock so no need to update anything on that front! Thanks!

**Comment by mcoker** [comment]
@mmenestr @mcarrano thanks! Designs LGTM.

**Comment by mcarrano** [comment]
Thanks for your efforts on this @mmenestr . I've opened https://github.com/patternfly/patternfly/issues/3380 to track core updates and will get this into a future milestone.
